### PR TITLE
Paginator division by zero warning

### PR DESCRIPTION
**Pull request by ruudboon**
Hello!

* Type: bug fix
* Link to issue: #14303

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [x] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Small description of change:

Fixed `Phalcon\Paginator\Adapter\AbstractAdapter` throw exception if `limit <= 0`

Thanks

### CODE REVIEW

**Review by niden — DISMISSED** [review]

File: tests/integration/Paginator/Adapter/QueryBuilder/GetSetLimitCest.php [hunk 1]
```diff
@@ -68,5 +69,21 @@ public function paginatorAdapterQuerybuilderGetSetLimit(IntegrationTester $I)
...
+                    'builder' => $builder,
+                    'limit'   => 0,
+                    'page'    => 1,
+                ]
+            );
+
+            $paginator->paginate();
+        } catch (Exception $ex) {
+            $I->assertEquals('Limit must be greater then zero', $ex->getMessage());
+        }
```
Comment (sergeyklay) [inline tests/integration/Paginator/Adapter/QueryBuilder/GetSetLimitCest.php]: If you want to check message or exception code, you can pass them with exception instance:
```php
// will check that exception MyException is thrown with "Don't do bad things" message
$I->expectException(new MyException("Don't do bad things"), function() {
    $this->doSomethingBad();
});
```

**Review by sergeyklay — CHANGES_REQUESTED** [review]

File: tests/integration/Paginator/Adapter/QueryBuilder/GetSetLimitCest.php (hunk 1 above)
Comment (sergeyklay) [reply tests/integration/Paginator/Adapter/QueryBuilder/GetSetLimitCest.php]: There is also `expectThrowable`. It handles and checks throwables (Exceptions/Errors) called inside the callback function. Either throwable class name or throwable instance should be provided:

```php
$I->expectThrowable(MyThrowable::class, function() {
    $this->doSomethingBad();
});

$I->expectThrowable(new MyException(), function() {
    $this->doSomethingBad();
});
```

Comment (sergeyklay) [reply tests/integration/Paginator/Adapter/QueryBuilder/GetSetLimitCest.php]: Do not use try/catch in tests. If `try` will not raise exception test always will pass, isn't? ;)

Comment (ruudboon) [reply tests/integration/Paginator/Adapter/QueryBuilder/GetSetLimitCest.php]: Phpstorm noticed that 'expectException' is deprecated so I tried a different method. Didn't know you could use 'expectThrowable'.

File: tests/integration/Paginator/Adapter/QueryBuilder/GetSetLimitCest.php [hunk 2]
```diff
@@ -68,5 +69,17 @@ public function paginatorAdapterQuerybuilderGetSetLimit(IntegrationTester $I)
         );
 
         $I->assertEquals($paginator, $setterResult);
+
+        $I->expectThrowable(new Exception('Limit must be greater then zero'), function() use ($builder) {
```
Comment (sergeyklay) [inline tests/integration/Paginator/Adapter/QueryBuilder/GetSetLimitCest.php]:
```
FILE: ...alcon/tests/integration/Paginator/Adapter/QueryBuilder/GetSetLimitCest.php

--------------------------------------------------------------------------------

FOUND 1 ERROR AFFECTING 1 LINE

--------------------------------------------------------------------------------

 73 | ERROR | [x] Expected 1 space after FUNCTION keyword; 0 found

    |       |     (Squiz.Functions.MultiLineFunctionDeclaration.SpaceAfterFunction)

--------------------------------------------------------------------------------

PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY

--------------------------------------------------------------------------------
```

**Review by sergeyklay — CHANGES_REQUESTED** [review]

**Review by sergeyklay — APPROVED** [review]

**Review by niden — APPROVED** [review]

**Pull request merged**